evm/src/main/java/org/hyperledger/besu/evm/operation/AbstractCallOperation.java

```diff
@@ -191,24 +198,16 @@ public OperationResult execute(final MessageFrame frame, final EVM evm) {
 
     final Account contract = frame.getWorldUpdater().get(to);
 
-    if (contract != null && contract.hasDelegatedCode()) {
-      if (contract.getCodeDelegationTargetCode().isEmpty()) {
-        throw new RuntimeException("A delegated code account must have delegated code");
-      }
-
-      if (contract.getCodeDelegationTargetHash().isEmpty()) {
-        throw new RuntimeException("A delegated code account must have a delegated code hash");
-      }
-
-      final long codeDelegationResolutionGas =
-          CodeDelegationGasCostHelper.codeDelegationGasCost(frame, gasCalculator(), contract);
-
-      if (frame.getRemainingGas() < codeDelegationResolutionGas) {
-        return new Operation.OperationResult(
-            codeDelegationResolutionGas, ExceptionalHaltReason.INSUFFICIENT_GAS);
-      }
-
-      frame.decrementRemainingGas(codeDelegationResolutionGas);
+    try {
+      deductGasForCodeDelegationResolution(frame, contract);
```

Comment:
is it normal to do this check before checking balance ? https://github.com/hyperledger/besu/pull/8408/files#diff-38fea70dec4587d8c58b86db43513592391fc36652e9b16c544555419625e176R217

Reply:
I moved the check now after the balance check.

Reply:
@matkt I needed to move the check back again, because it was not spec compliant anymore. The issue is that we need to check if the sender has enough balance before we check for message frame too deep. In [line 217](https://github.com/hyperledger/besu/blob/99e53e25449e8ce3d1b399bdfa44e6cfaf5462ca/evm/src/main/java/org/hyperledger/besu/evm/operation/AbstractCallOperation.java#L217) we check the balance and message frame depth at the same time, so the gas deduction for 7702 code resolution needs to be before that.